include/wil/cppwinrt_helpers.h

```diff
@@ -312,6 +316,229 @@ namespace wil
+        {
+        }
+#endif
+
+        uint32_t produce_values(winrt::array_view<TResult> const& view)
+        {
+            if (m_status == IterationStatus::Producing)
+            {
+                return 0;
+            }
+            else if (m_status == IterationStatus::Done)
+            {
+                throw winrt::hresult_out_of_bounds();
+            }
+
+            m_values = view;
+            m_current = m_values.begin();
+            m_status = IterationStatus::Producing;
+
+            coroutine_handle<iterator_promise>::from_promise(*this).resume();
+
+            return static_cast<uint32_t>(m_current - m_values.begin());
+        }
+
+        bool HasCurrent() const noexcept
+        {
+            return m_status == IterationStatus::Value;
+        }
+
+        TResult Current() const noexcept
+        {
+            return m_last_value;
+        }
+
+        uint32_t GetMany(winrt::array_view<TResult> values)
+        {
+            if (!HasCurrent() || values.empty())
+            {
+                return 0;
+            }
+
+            values.front() = Current();
+
+            uint32_t result;
+
+            if (values.size() == 1)
+            {
+                result = 1;
+            }
+            else
+            {
+                result = produce_values({ values.data() + 1, values.size() - 1 }) + 1;
+            }
+
+            MoveNext();
+            return result;
+        }
+
+        bool MoveNext()
+        {
+            return produce_values({ &m_last_value, 1 });
+        }
+
+    private:
+        IterationStatus m_status{ IterationStatus::Producing };
+        winrt::array_view<TResult> m_values{ &m_last_value, 1 };
+        TResult* m_current{ &m_last_value };
+        TResult m_last_value{ empty<TResult>() };
+    };
+
+    template<typename TResult, typename Func, typename... Args>
+    struct iterable_iterator_helper : winrt::implements<
+            iterable_iterator_helper<TResult, Func, Args...>,
+            winrt::Windows::Foundation::Collections::IIterable<TResult>
+            >
+    {
+        iterable_iterator_helper(Func&& func, Args&&... args) :
+            m_func{ std::forward<Func>(func) },
+            m_args{ std::forward<Args>(args)... }
+        {
+        }
+
+        auto First()
+        {
+            return std::apply(m_func, m_args);
+        }
+
+    private:
+        Func m_func;
+        std::tuple<Args...> m_args;
+    };
+
+    template<typename>
+    struct iterator_result;
+
+    template<typename TResult>
+    struct iterator_result<winrt::Windows::Foundation::Collections::IIterator<TResult>>
+    {
+        using type = TResult;
+    };
+}
+/// @endcond
+
+namespace wil
+{
+    template<typename Func, typename... Args, typename TResult = typename details::iterator_result<std::invoke_result_t<Func, Args...>>::type>
+    winrt::Windows::Foundation::Collections::IIterable<TResult> make_iterable_from_iterator(Func&& func, Args&&... args)

```

Comment:
```suggestion
    winrt::Windows::Foundation::Collections::IIterable<TResult> make_iterable_from_generator(Func&& func, Args&&... args)
```
Since you're creating from a function (a generator) and not an `IIterable` itself